File: NET_8/FoodCall.Domain/Exceptions/DuplicateEntityException.cs
namespace FoodCall.Domain.Exceptions;

public class DuplicateEntityException : DomainException
{
    public DuplicateEntityException(string entityName, string identifier) 
        : base($"{entityName} com '{identifier}' já existe no sistema.")
    {
    }

    public DuplicateEntityException(string message) 
        : base(message)
    {
    }
}

File: Java/src/main/java/com/foodcall/domain/exceptions/DuplicateEntityException.java
package com.foodcall.domain.exceptions;

public class DuplicateEntityException extends DomainException {
    
    public DuplicateEntityException(String entityName, String identifier) {
        super(String.format("%s com '%s' já existe no sistema.", entityName, identifier));
    }
    
    public DuplicateEntityException(String message) {
        super(message);
    }
}